units/crons.go

```diff
@@ -1020,6 +1020,13 @@ func PopulateCacheHistoricalTestDataJob(part int) amboy.QueueOperation {
 	}
 }
 
+func PopulateSpawnhostExpirationCheckJob() amboy.QueueOperation {
```

Comment:
I think you can optimize this by doing the FindHostsWithNoExpirationToExtend check here, and only running the job if there are any (maybe also pass the hosts down to the job once you've found them)

Reply:
I opted to just create a new job for every host we need to extend -- dealing with passing slices of IDs of hosts to the job seemed too messy, but let me know if this is also an inefficient way to use jobs.